modules/aggs-matrix-stats/src/main/java/org/opensearch/search/aggregations/matrix/stats/MatrixStatsAggregationBuilder.java

```diff
@@ -74,22 +75,19 @@ protected AggregationBuilder shallowCopy(AggregatorFactories.Builder factoriesBu
      */
     public MatrixStatsAggregationBuilder(StreamInput in) throws IOException {
         super(in);
+        this.multiValueMode = in.readEnum(MultiValueMode.class);
```

Comment:
This won't be backwards compatible unless you put version guards around the reads and writes to only expect the new data on or after version 3.1

Reply:
Thanks! I've added version guards as suggested here: ([eb4ac67](https://github.com/opensearch-project/OpenSearch/pull/18254/commits/eb4ac67857ffb87308a2829b1e01082a6f9c0045))